bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/protocol/philipstv/ConnectionManager.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.androidtv.internal.protocol.philipstv;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+import org.apache.http.HttpHost;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpResponseException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.conn.HttpHostConnectException;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * The {@link ConnectionManager} is responsible for handling https GETs and POSTs to the Philips
+ * TVs.
+ *
+ * @author Benjamin Meyer - Initial contribution
+ * @author Ben Rosenblum - Merged into AndroidTV
+ */
+@NonNullByDefault
+public class ConnectionManager {
+
+    private static final String TARGET_URI_MSG = "Target Uri is: {}";
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    public static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    // Cannot use jetty in OH2.4 due to 9.4.11.v20180605 version with digest auth bug
+    // https://github.com/eclipse/jetty.project/issues/1555
+    private final CloseableHttpClient httpClient;
```

Comment:
The related issue was closed 5+ years ago. And not long ago openHAB upgraded to Jetty 9.4.52.v20230823 Would it be possible to use the common client?

Reply:
I suppose that's definitely possible.  I was trying to avoid as much rewrite as possible on this pass through.  It's my intention to try to optimize this code base later.